M2/Macaulay2/m2/matrix.m2

```diff
@@ -116,9 +118,12 @@ Matrix == Matrix := (f,g) -> (
            raw(super f * inducedMap(source f, source g)) === raw super g
     )
 
-Matrix == Number :=
-Matrix == RingElement := (m,f) -> m - f == 0		    -- slow!
+Matrix == Number      :=
+Matrix == RingElement :=
+Vector == Number      :=
+Vector == RingElement := (m,f) -> m - f == 0                -- slow!
```

Comment:
Does this need to be written this way? This is slow due to #3035. For instance try `v == 1` with `v` the matrix defined in that issue.

Reply:
Also, `Vector == Number` covers `Vector == ZZ`, no?

Reply:
Do you have a suggestion for a better way to define `Matrix == Number` and friends?   That's probably outside the scope of this PR.

Regarding `Vector == ZZ`, we currently need it to avoid a recursion error.  Otherwise, something like `v == 0` would call `v - 0 == 0`.

Reply:
I guess I'm wondering what's wrong with `m == matrix f`? The slight edge case is when the target of m has relations and you want to reduce f first, but maybe the user should explicitly write `m - f * id_(target m)` in that case.

Reply:
Sure, but this doesn't apply to vectors.

Reply:
Calling `matrix` on the `Number`/`RingElement` won't work for the zero vector if the rank of the module is greater than one, since then we'll be comparing matrices with different sizes:

```m2
i1 : matrix vector {0, 0} == matrix 0

o1 = false
```
I did change it to just `matrix v == f`, which works just fine:

```m2
i2 : matrix vector {0, 0} == 0

o2 = true
```
Then we don't have to worry about defining `Vector == ZZ`.